linalg/operator.hpp

```diff
@@ -384,62 +392,87 @@ class TimeDependentOperator : public Operator
+       consider the backward Euler method defined by
+       $ y(t + \Delta t) = y(t) + \Delta t k_0 $ where
+       $ M k_0 = g \big( y(t) + \Delta t k_0 \big) $. A backward Euler
+       integrator can use @a k from this function, with @a u set to $ y(t) $ and
+       @a gamma set to $ \Delta t$, for $k_0$. Generalizing further, consider a
+       diagonally implicit Runge-Kutta (DIRK) method defined by
+       $ y(t + \Delta t) = y(t) + \Delta t \sum_{i=1}^s b_i k_i $ where
+       $ M k_i = g \big( y(t) + \Delta t \sum_{j=1}^i a_{ij} k_j, t + c_i \Delta t \big) $.
+       A DIRK integrator can use @a k from this function, with @a u set to
+       $ y(t) + \Delta t \sum_{j=1}^{i-1} a_{ij} k_j $ and @a gamma set to
+       $ a_{ii} \Delta t $, for $ k_i $.
 
        If not re-implemented, this method simply generates an error. */
-   virtual void ImplicitSolve(const double dt, const Vector &x, Vector &k);
+   virtual void ImplicitSolve(const double gamma, const Vector &u, Vector &k);
 
-   /** @brief Return an Operator representing (dF/dk @a shift + dF/dx) at the
-       given @a x, @a k, and the currently set time.
+   /** @brief Return an Operator representing (dF/dk @a shift + dF/du) at the
+       given @a u, @a k, and the currently set time.
 
        Presently, this method is used by some PETSc ODE solvers, for more
        details, see the PETSc Manual. */
-   virtual Operator& GetImplicitGradient(const Vector &x, const Vector &k,
+   virtual Operator& GetImplicitGradient(const Vector &u, const Vector &k,
                                          double shift) const;
 
-   /** @brief Return an Operator representing dG/dx at the given point @a x and
+   /** @brief Return an Operator representing dG/du at the given point @a u and
        the currently set time.
 
        Presently, this method is used by some PETSc ODE solvers, for more
        details, see the PETSc Manual. */
-   virtual Operator& GetExplicitGradient(const Vector &x) const;
-
-   /** @brief Setup the ODE linear system $ A(x,t) = (I - gamma J) $ or
-       $ A = (M - gamma J) $, where $ J(x,t) = \frac{df}{dt(x,t)} $.
-
-       @param[in]  x     The state at which $A(x,t)$ should be evaluated.
-       @param[in]  fx    The current value of the ODE rhs function, $f(x,t)$.
+   virtual Operator& GetExplicitGradient(const Vector &u) const;
+
+   /** @brief For solving an ordinary differential equation of the form
+       $ M \frac{dy}{dt} = g(y,t) $, recall F and G are defined as one of
+       the following:
+       1. F(u,k,t) = k and G(u,t) = inv(M) g(u,t)
+       2. F(u,k,t) = M k and G(u,t) = g(u,t)
+       3. F(u,k,t) = M k - g(u,t) and G(u,t) = 0
+       This function performs setup to solve $ A x = b $ where A is either
+       1. A(@a y,t) = I - @a gamma inv(M) J(@a y,t)
+       2. A(@a y,t) = M - @a gamma J(@a y,t)
+       3. A(@a y,t) = M - @a gamma J(@a y,t)
+       with J = dg/dy (or a reasonable approximation thereof).
```

Comment:
For correct Doxygen formatting and improved editor formatting, I suggest:
```suggestion
   /** @brief Setup a linear system as needed by some SUNDIALS ODE solvers.

       For solving an ordinary differential equation of the form
       $ M \frac{dy}{dt} = g(y,t) $, recall that F and G can be defined as one
       of the following:

         1. F(u,k,t) = k and G(u,t) = inv(M) g(u,t)
         2. F(u,k,t) = M k and G(u,t) = g(u,t)
         3. F(u,k,t) = M k - g(u,t) and G(u,t) = 0

       This function performs setup to solve $ A x = b $ where A is either

         1. A(@a y,t) = I - @a gamma inv(M) J(@a y,t)
         2. A(@a y,t) = M - @a gamma J(@a y,t)
         3. A(@a y,t) = M - @a gamma J(@a y,t)

       with J = dg/dy (or a reasonable approximation thereof).
```